src/Uno.UI/UI/Xaml/UnsetValue.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+
+namespace Windows.UI.Xaml
+{
+	/// <summary>
+	/// This class is used for DependencyProperty.UnsetValue.
+	/// </summary>
+	internal class UnsetValue : IEquatable<UnsetValue>
+	{
+		private UnsetValue()
+		{
+		}
+
+		internal static UnsetValue Instance { get; } = new UnsetValue();
+
+		public bool Equals(UnsetValue other) => !(other is null);
+
+		public override bool Equals(object obj)
+		{
+			return obj is UnsetValue;
```

Comment:
No sure if a `ReferenceEqual` is better here...

Reply:
Even if it were, there should not be that may comparisons done that way.

Reply:
I assume that the runtime will try to perform a type conversion, so yeah I think that ref equals is probably faster